src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/MutableObjectModelBinder.cs

```diff
@@ -157,9 +214,17 @@ private bool CanValueBindAnyModelProperties(MutableObjectBinderContext context)
             // create the model and try to bind it. OR if ALL properties of the model have a greedy source,
             // then we go ahead and create it.
             //
+            var hasBindableProperty = false;
             var isAnyPropertyEnabledForValueProviderBasedBinding = false;
-            foreach (var propertyMetadata in context.PropertyMetadata)
+            foreach (var propertyMetadata in bindingContext.ModelMetadata.Properties)
             {
+                if (!CanBindProperty(bindingContext, propertyMetadata))
```

Comment:
`CanBindProperty` is cheap to execute, no real reason to cache the result of it like the previous code did.